### PR TITLE
feat: show usd price with no input amount

### LINKED ISSUE #1712: Show spot price on token select
We should consider fetching and displaying the spot price of the pair as soon as a token is selected rather than relying on the use to input a value. This will be useful for people who are checking the current market price of a token pair and reduce clicks/actions.

### PR DESCRIPTION
**Pull request by just-toby**
## Description

fixes #1712 by defaulting to an input amount of "1" when calculating the USD value

_Linear ticket:_ https://github.com/Uniswap/interface/issues/1712

### Before
image

### After

image

## Test plan

Screenshot 2023-09-18 at 10 11 57 AM

### CODE REVIEW

File: src/pages/Swap/index.tsx [hunk 1]
```diff
@@ -314,10 +314,20 @@ export function Swap({
...
-  const fiatValueInput = useUSDPrice(parsedAmounts[Field.INPUT], currencies[Field.INPUT] ?? undefined)
-  const fiatValueOutput = useUSDPrice(parsedAmounts[Field.OUTPUT], currencies[Field.OUTPUT] ?? undefined)
-  const showFiatValueInput = Boolean(parsedAmounts[Field.INPUT])
-  const showFiatValueOutput = Boolean(parsedAmounts[Field.OUTPUT])
+  const fiatValueInput = useUSDPrice(
+    parsedAmounts[Field.INPUT] ??
+      (currencies[Field.INPUT]
+        ? CurrencyAmount.fromRawAmount(currencies[Field.INPUT], JSBI.BigInt(10 ** currencies[Field.INPUT].decimals))
+        : undefined),
+    currencies[Field.INPUT] ?? undefined
```
Comment (natew) [inline src/pages/Swap/index.tsx:322]: `?? undefined` i think that can just be removed

Comment (natew) [reply src/pages/Swap/index.tsx:322]: unless currencies can return `null` and it only wants an undefined type there which would be odd

**Review by natew — CHANGES_REQUESTED** [review]
see comment

File: src/pages/Swap/index.tsx [hunk 2]
```diff
@@ -314,10 +314,20 @@ export function Swap({
...
+        : undefined),
+    currencies[Field.INPUT]
+  )
+  const fiatValueOutput = useUSDPrice(
+    parsedAmounts[Field.OUTPUT] ??
+      (currencies[Field.OUTPUT]
+        ? CurrencyAmount.fromRawAmount(currencies[Field.OUTPUT], JSBI.BigInt(10 ** currencies[Field.OUTPUT].decimals))
+        : undefined),
+    currencies[Field.OUTPUT]
+  )
```
Comment (JFrankfurt) [inline src/pages/Swap/index.tsx:330]: this is muuuuch harder to parse quickly. Can you find a more readable way to break out the same logic?

**Review by JFrankfurt — APPROVED** [review]

File: src/state/swap/hooks.tsx [hunk 3]
```diff
@@ -77,7 +77,7 @@ const BAD_RECIPIENT_ADDRESSES: { [address: string]: true } = {
 }
 
 export type SwapInfo = {
-  currencies: { [field in Field]?: Currency | null }
+  currencies: { [field in Field]?: Currency | undefined }
```
Comment (natew) [inline src/state/swap/hooks.tsx:80]: `?:` already means `| undefined` so no need for `| undefined`

File: src/lib/hooks/useCurrency.ts [hunk 4]
```diff
@@ -116,7 +116,7 @@ export function useCurrencyFromMap(
 
   const token = useTokenFromMapOrNetwork(tokens, isNative ? undefined : shorthandMatchAddress ?? currencyId)
 
-  if (currencyId === null || currencyId === undefined || !isSupportedChain(chainId)) return null
+  if (currencyId === null || currencyId === undefined || !isSupportedChain(chainId)) return undefined
```
Comment (natew) [inline src/lib/hooks/useCurrency.ts:119]: can just `return` as it returns undefined

**Review by natew — CHANGES_REQUESTED** [review]
small changes

**Review by cbachmeier — APPROVED** [review]

**Pull request merged**